File: db/country.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in country.
*/

class Country extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "country";
    $this->order = "";
    $this->domain = "valuelist";
    $this->field['country_id']['type'] = 'SERIAL';
    $this->field['country_id']['not_null'] = true;
    $this->field['iso_3166_code']['type'] = 'CHAR';
    $this->field['iso_3166_code']['length'] = 2;
    $this->field['iso_3166_code']['not_null'] = true;
    $this->field['phone_prefix']['type'] = 'VARCHAR';
    $this->field['phone_prefix']['length'] = 8;
    $this->field['f_visible']['type'] = 'BOOL';
    $this->field['f_visible']['not_null'] = true;
    $this->field['f_visible']['default'] = true;
    $this->field['f_preferred']['type'] = 'BOOL';
    $this->field['f_preferred']['not_null'] = true;
    $this->field['f_preferred']['default'] = true;
    $this->field['country_id']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/country.rb
module Momomoto
  class Country < Base
    def initialize
      super
      @domain = 'valuelist'
      @fields = {
        :country_id => Datatype::Integer.new( {:not_null=>true, :default=>true, :primary_key=>true, :serial=>true} ),
        :iso_3166_code => Datatype::Char.new( {:length=>2, :not_null=>true} ),
        :phone_prefix => Datatype::Varchar.new( {:length=>8} ),
        :f_visible => Datatype::Bool.new( {:not_null=>true, :default=>true} ),
        :f_preferred => Datatype::Bool.new( {:not_null=>true, :default=>true} )
      }
    end
  end
end
